.github/workflows/build-and-push.yml

```diff
@@ -20,29 +21,64 @@ jobs:
       run: sudo ./.github/install-etcd.sh
     - name: Build with Maven
       run: mvn -B package --file pom.xml
+
   build:
     needs: test
     runs-on: ubuntu-latest
+
     env:
-      IMAGE_NAME: kserve/modelmesh
+      IMAGE_NAME: modelmesh
+
     steps:
-    - uses: actions/checkout@v2
-    - name: Build and push runtime image
-      run: |
-        GIT_COMMIT=$(git rev-parse HEAD)
-        BUILD_ID=$(date '+%Y%m%d')-$(git rev-parse HEAD | cut -c -5)
+      - uses: actions/checkout@v2
+
+      - name: Setup QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Setup Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Login to DockerHub
+        if: github.event_name == 'push'
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USER }}
+          password: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+
+      - name: export version variable
+        run: |
+          GIT_COMMIT=$(git rev-parse HEAD)
+          BUILD_ID=$(date '+%Y%m%d')-$(git rev-parse HEAD | cut -c -5)
+          
+          IMAGE_ID=kserve/$IMAGE_NAME
+
+          # Change all uppercase to lowercase
+          IMAGE_ID=$(echo $IMAGE_ID | tr '[A-Z]' '[a-z]')
+
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
 
-        # Strip git ref prefix from version
-        VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name
+          # [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
 
-        # Use Docker `latest` tag convention
-        [ "$VERSION" == "main" ] && VERSION=latest
-        echo $VERSION
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "main" ] && VERSION=latest
 
-        docker build -t ${{ env.IMAGE_NAME }}:$VERSION \
-          --build-arg imageVersion=$VERSION \
-          --build-arg buildId=${BUILD_ID} \
-          --build-arg commitSha=${GIT_COMMIT} .
+          echo "GIT_COMMIT=$GIT_COMMIT" >> $GITHUB_ENV
+          echo "BUILD_ID=$BUILD_ID" >> $GITHUB_ENV
+          echo "VERSION=$VERSION" >> $GITHUB_ENV
+          echo "IMAGE_ID=$IMAGE_ID" >> $GITHUB_ENV
 
-        docker login -u ${{ secrets.DOCKER_USER }} -p ${{ secrets.DOCKER_ACCESS_TOKEN }}
-        docker push ${{ env.IMAGE_NAME }}:$VERSION
+      - name: Build and push
+        uses: docker/build-push-action@v3
+        with:
+          platforms: linux/amd64,linux/arm64/v8,linux/ppc64le
+          pull: true
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          push: ${{ github.event_name == 'push' }}
+          tags: ${{ env.IMAGE_ID }}:${{ env.VERSION }}
+          build-args: |
```

Comment:
we may need another variable / docker build `ARG` for the `etcd` installation inside the Docker image

https://docs.docker.com/build/building/multi-platform/#building-multi-platform-images

> ... A list of build arguments like `BUILDPLATFORM` and `TARGETPLATFORM` is available automatically inside your Dockerfile and can be leveraged by the processes running as part of your build.

https://blog.thesparktree.com/docker-multi-arch-github-actions

> `TARGETARCH` - architecture component of `TARGETPLATFORM`


https://github.com/kserve/modelmesh/blob/38c1385b7a046c73f5097ef3c0c57a4e35099fd8/Dockerfile#L36-L39

... `etcd-${ETCD_VERSION}-linux-amd64.tar.gz` should be
... `etcd-${ETCD_VERSION}-linux-${TARGETARCH}.tar.gz`

Assets listed on the corresponding release page for `etcd`:
 https://github.com/etcd-io/etcd/releases/tag/v3.5.4

- etcd-v3.5.4-linux-**amd64**.tar.gz
- etcd-v3.5.4-linux-**arm64**.tar.gz
- etcd-v3.5.4-linux-**ppc64le**.tar.gz
- etcd-v3.5.4-linux-**s390x**.tar.gz

Default in the Dockerfile should probably be:

```Dockerfile
ARG TARGETARCH=amd64
```

Reply:
ah good catch, on it